src/inmanta/ast/type.py

```diff
@@ -418,9 +550,22 @@ def type_string(self) -> str:
     def type_string_internal(self) -> str:
         return "List"
 
-    def get_location(self) -> None:
+    def get_location(self) -> None | Location:
         return None
 
+    def corresponds_to(self, type: Type) -> bool:
+        if isinstance(type, Any):
+            return True
+
+        return isinstance(type, List)
```

Comment:
Do we want to allow children here (i.e. `TypedList`)? I would expect `return builtins.type(type) is self.__class__ and self.base_type() is self or self.base_type().corresponds_to(type.base_type())` or something (everything from the first `and` on is to cover the child implementation, might not be appropriate here)

Reply:
The model can't produce this type, so it is a moot point. I could just assert False here I think. 

Would it be better to make it raise an exception when called?

Reply:
Right, when I typed it I was thinking it was the DSL's `list` type, but that's `TypedList(Literal())`. I guess it's fine like this then. Perhaps add a comment that it's currently not expected to be called?